libraries/node-core-library/src/FileSystem.ts

```diff
@@ -829,25 +892,98 @@ export class FileSystem {
+      return fsx.realpathSync(linkPath);
+    });
   }
 
   /**
    * An async version of {@link FileSystem.getRealPath}.
    */
   public static getRealPathAsync(linkPath: string): Promise<string> {
-    return fsx.realpath(linkPath);
+    return FileSystem._wrapExceptionAsync(() => {
+      return fsx.realpath(linkPath);
+    });
   }
 
   // ===============
   // UTILITY FUNCTIONS
   // ===============
 
   /**
-   * Returns true if the error provided indicates the file or folder
-   * does not exist.
+   * Returns true if the error provided indicates the file or folder does not exist.
    */
   public static isNotExistError(error: NodeJS.ErrnoException): boolean {
-    return error.code === 'ENOENT' || error.code === 'ENOTDIR';
+    return FileSystem.isFileDoesNotExistError(error) || FileSystem.isFolderDoesNotExistError(error);
+  }
+
+  /**
+   * Returns true if the error provided indicates the file does not exist.
+   */
+  public static isFileDoesNotExistError(error: NodeJS.ErrnoException): boolean {
+    return FileSystem._isErrorType(error) && (error.code === 'ENOENT');
+  }
+
+  /**
+   * Returns true if the error provided indicates the folder does not exist.
+   */
+  public static isFolderDoesNotExistError(error: NodeJS.ErrnoException): boolean {
+    return FileSystem._isErrorType(error) && (error.code === 'ENOTDIR');
+  }
+
+  private static _isErrorType(error: NodeJS.ErrnoException): boolean {
+    if (process.env.NODE_ENV !== 'test' && process.env.JEST_WORKER_ID === undefined) {
+      return error instanceof Error;
+    } else {
+      // Using this instead of `error instanceof Error` because that doesn't work in Jest tests
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return (error as any).__proto__ === (FileSystem._fsError as any).__proto__;
+    }
+  }
+
+  private static get _fsError(): NodeJS.ErrnoException {
+    if (!FileSystem.__error) {
+      try {
+        fsx.readFileSync(nodeJsPath.join(__dirname, 'fileThatDoesn\'tExist'));
+      } catch (error) {
+        FileSystem.__error = error;
+      }
+    }
+
+    return FileSystem.__error;
+  }
+
+  private static _wrapException<TResult>(fn: () => TResult): TResult {
+    function updateErrorMessage(error: NodeJS.ErrnoException): void {
+      if (FileSystem.isFileDoesNotExistError(error)) { // eslint-disable-line @typescript-eslint/no-use-before-define
+        error.message = `File does not exist: ${error.path}`;
+      } else if (FileSystem.isFolderDoesNotExistError(error)) { // eslint-disable-line @typescript-eslint/no-use-before-define
+        error.message = `Folder does not exist: ${error.path}`;
+      }
+    }
+
+    try {
+      return fn();
+    } catch (error) {
+      updateErrorMessage(error);
+      throw error;
+    }
+  }
+
+  private static async _wrapExceptionAsync<TResult>(fn: () => Promise<TResult>): Promise<TResult> {
+    function updateErrorMessage(error: NodeJS.ErrnoException): void {
```

Comment:
Given that this code is copy+pasted in two places, what is the point of making it a local function?

I feel like `updateErrorMessage()` should be a static private member of `FileSystem`.